test/data/ansible/check_ostree.yaml

```diff
@@ -342,6 +343,36 @@
       shell: dmesg --notime | grep -i "error\|fail" || true
       register: result_dmesg_error
 
+    - name: check embeded container image with podman
+      command: podman images
+      become: yes
+      register: result_podman_images
+      when:
+        - embeded_container == "true"
+        - (ansible_facts['distribution'] == 'RedHat' and ansible_facts ['distribution_version'] is version('9.0', '>')) or
+          (ansible_facts['distribution'] == 'RedHat' and ansible_facts ['distribution_version'] is version('8.6', '>') and ansible_facts ['distribution_version'] is version('9.0', '!=')) or
+          (ansible_facts['distribution'] == 'CentOS')
```

Comment:
If the variable is set correctly from the calling script there is no need to duplicate the conditions on distro name & version here.